README.md

```diff
@@ -301,5 +337,30 @@ format and focus on more important stuff.
       default: {
         assert(false);
       }
-    }
+    } 
     ```
+
+1. Use smart_pointer as much as you can. Try unique_ptr rather than shared_ptr.
+
+   * Smart Pointer is a better way to handle memory alloc/free then manually new/delete.
+      * especially in multi-thread situation.
+   * unique_ptr is much lighter than shared_ptr.
+      * shared_ptr maintain a mutex, a reference count. Create and destory a shared_ptr could be a little bit slower than unique_ptr.
+      * unique_ptr is better for multi-thread condition. It ensures that there are no other thread using this data right now.
+   * Use unique_ptr<int[] > rather than vector<int> for array that will not grow.
+      * unique_ptr is easier to return.
+   * Use weak_ptr to avoid circle reference in shared_ptr.
+
+1. Use **const** as much as you can. With C++11, `constexpr` is a
+    better choice for some uses of `const`
+
+
+   * It will let compiler to check whether program is correct or not, rather than die in runtime.
```

Comment:
s/to//

I think we need some context on this to make it easier to understand. Care to share a code example?